ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -715,13 +716,17 @@ protected void maybeProcessCallbacks() {
         // Only when the blob is encrypted should we need to call this method. When finish decryption, we don't need
         // response info anymore.
         if (decryptCallbackResultInfo.exception == null) {
-          chunkIndexToBuf.put(chunkIndex,
-              filterChunkToRange(decryptCallbackResultInfo.result.getDecryptedBlobContent()));
-          numChunksRetrieved.incrementAndGet();
+          DecryptJob.DecryptJobResult result = null;
```

Comment:
nit, it may be a bit more readable to do the getandset on this line:
```java
DecryptJob.DecryptJobResult result = decryptCallbackResultInfo.result.getAndSet(null);
if (result != null) {...
```